libcurl 7.54 build failure

While building it on an oldish macOS, I got the error below. This macOS comes with libcurl/7.54.0, which according to your readme it's supported ("Minimum supported libcurl version : 7.28.0").

Using the homebrew libcurl, which is currently 7.80, it works. I had to manually set
```
export C_INCLUDE_PATH=/usr/local/Cellar/curl/7.80.0_1/include/
export LIBRARY_PATH=/usr/local/Cellar/curl/7.80.0_1/lib/:$LIBRARY_PATH
```

```
#=== ERROR while compiling ocurl.0.9.2 ========================================#
# context     2.1.1 | macos/x86_64 | ocaml.4.12.0 | https://opam.ocaml.org#f797da1a
# path        ~/.opam/default/.opam-switch/build/ocurl.0.9.2
# command     ~/.opam/opam-init/hooks/sandbox.sh build make
# exit-code   2
# env-file    ~/.opam/log/ocurl-67741-fd92ab.env
# output-file ~/.opam/log/ocurl-67741-fd92ab.out
### output ###
# curl-helper.c:1394:9: warning: implicit declaration of function 'curl_global_sslset' is invalid in C99 [-Wimplicit-function-declaration]
# [...]
#         ^
# curl-helper.c:1396:7: error: use of undeclared identifier 'res'
#   if (res != CURLSSLSET_UNKNOWN_BACKEND)
#       ^
# curl-helper.c:1396:14: error: use of undeclared identifier 'CURLSSLSET_UNKNOWN_BACKEND'
#   if (res != CURLSSLSET_UNKNOWN_BACKEND)
#              ^
# fatal error: too many errors emitted, stopping now [-ferror-limit=]
# 1 warning and 20 errors generated.
# make[1]: *** [curl-helper.o] Error 2
# make: *** [all] Error 2
```

--- a/configure
+++ b/configure
@@ -8768,6 +8768,26 @@ $as_echo "#define HAVE_CURL_MULTI_POLL 1" >>confdefs.h
 fi
 
 
+ac_fn_c_check_decl "$LINENO" "curl_global_sslset" "ac_cv_have_decl_curl_global_sslset" "
+#include \"curl/curl.h\"
+
+"
+if test "x$ac_cv_have_decl_curl_global_sslset" = xyes; then :
+  ac_have_decl=1
+else
+  ac_have_decl=0
+fi
+
+cat >>confdefs.h <<_ACEOF
+#define HAVE_DECL_CURL_GLOBAL_SSLSET $ac_have_decl
+_ACEOF
+if test $ac_have_decl = 1; then :
+
+$as_echo "#define HAVE_CURL_GLOBAL_SSLSET 1" >>confdefs.h
+
+fi
+
+
 ac_config_headers="$ac_config_headers config.h"
 
 ac_config_files="$ac_config_files Makefile examples/Makefile"

--- a/configure.ac
+++ b/configure.ac
@@ -219,6 +219,14 @@ AC_DEFINE([HAVE_CURL_MULTI_POLL], [1], [Does libcurl define curl_multi_poll]),
 #include "curl/curl.h"
 ])
 
+AC_CHECK_DECLS(
+[curl_global_sslset],
+AC_DEFINE([HAVE_CURL_GLOBAL_SSLSET], [1], [Does libcurl define curl_global_sslset]),
+[],
+[
+#include "curl/curl.h"
+])
+
 AC_CONFIG_HEADERS(config.h)
 AC_CONFIG_FILES([Makefile examples/Makefile])
 

--- a/curl-helper.c
+++ b/curl-helper.c
@@ -1324,7 +1324,7 @@ curl_sslbackend sslBackendMap[] = {
 #endif
 };
 
-#if HAVE_DECL_CURLSSLBACKEND_NONE
+#if HAVE_CURL_GLOBAL_SSLSET
 /* Same order as in OCaml */
 CURLsslset sslsetMap[] = {
   CURLSSLSET_OK,

